packages/react-components/react-menu/src/components/Menu/useMenu.tsx

```diff
@@ -261,39 +251,29 @@ const useMenuOpenState = (
   }, []);
 
   // Manage focus for open state
-  const { findFirstFocusable, findNextFocusable, findPrevFocusable } = useFocusFinders();
+  const { findFirstFocusable } = useFocusFinders();
   const focusFirst = React.useCallback(() => {
     const firstFocusable = findFirstFocusable(state.menuPopoverRef.current);
     firstFocusable?.focus();
   }, [findFirstFocusable, state.menuPopoverRef]);
 
-  const focusAfterMenuTrigger = React.useCallback(() => {
-    const nextFocusable = findNextFocusable(state.triggerRef.current);
-    nextFocusable?.focus();
-  }, [findNextFocusable, state.triggerRef]);
-
-  const focusBeforeMenuTrigger = React.useCallback(() => {
-    const prevFocusable = findPrevFocusable(state.triggerRef.current);
-    prevFocusable?.focus();
-  }, [findPrevFocusable, state.triggerRef]);
-
   React.useEffect(() => {
     if (open) {
       focusFirst();
     } else {
       if (shouldHandleCloseRef.current) {
-        if (shouldHandleTabRef.current && !state.isSubmenu) {
-          pressedShiftRef.current ? focusBeforeMenuTrigger() : focusAfterMenuTrigger();
-        } else {
-          state.triggerRef.current?.focus();
-        }
+        // We know that React effects are sync so we focus the trigger here
+        // after any event handler (event handlers will call setState).
+        // Since the browser only performs the default behaviour for Tab once
+        // events have fully bubbled up the window, the browser will move
+        // focus to the next tabbable element before/after the trigger if needed.
+        // If tab was not pressed, then we simply focus the trigger as expected.
```

Comment:
```suggestion
        // We know that React effects are sync, so we focus the trigger here
        // after any event handler (event handlers will state updates and re-renders).
        // Since the browser only performs the default behaviour for Tab key once
        // keyboard events have fully bubbled up the window, the browser will move
        // focus to the next tabbable element before/after the trigger if needed.
        // If Tab key was not pressed, focus will remain on the trigger as expected.
```

nit: Tried to make more accurate

Reply:
```suggestion
        // We know that React effects are sync so we focus the trigger here
        // after any event handler (event handlers will update state and re-render).
        // Since the browser only performs the default behaviour for the Tab key once
        // keyboard events have fully bubbled up the window, the browser will move
        // focus to the next tabbable element before/after the trigger if needed.
        // If the Tab key was not pressed, focus will remain on the trigger as expected.
```